kubernetes/kubernetes_service.go

```diff
@@ -105,6 +105,18 @@ func (in *IstioClient) GetPods(namespace string, labelsSet labels.Set) (*v1.PodL
 	return in.k8s.CoreV1().Pods(namespace).List(meta_v1.ListOptions{LabelSelector: formatted})
 }
 
+// GetValidAppPods returns the pods definitions that have the labeling
+// to be considered as a service under an Istio context.
+// It returns an error on any problem.
+func (in *IstioClient) GetValidAppPods(namespace string) (*v1.PodList, error) {
```

Comment:
Do we really need this specialized method or could you have called the existing GetPods and passed in the desired label set?

Reply:
IMO, we need it.
I tried passing labels as:
```go
labels.Set{
  "app": "*"
  "version": "*"
}
```
but kubernetes complains. It doesn't allow * character (makes sense, this is not prometheus). I also tried empty strings instead of  *, but k8s tries to match the labels with empty strings and no pods are returned.

So, I had two choices: write the specialized method or write a method that can receive the "raw" label selector.

I chose the former since I see that the pattern in our client is to not allow the raw selectors.

Reply:
@israel-hdez I'm surprised it doesn't work with empty strings. Looking at this test file: `vendor/k8s.io/apimachinery/pkg/labels/selector_test.go` func `TestSelectorMatches` it seems that it should work:

`expectMatch(t, "x", Set{"x": "z"})` line 112

Reply:
Ok, I'll go with @jotak suggestion and change `GetPods` signature. It's what makes more sense to me.

Reply:
@jotak @lucasponce Signature it's changed. I hope it's ok now.

Reply:
thanks @israel-hdez !